src/sentry/search/events/datasets/discover.py

```diff
@@ -256,6 +266,19 @@ def function_converter(self) -> Mapping[str, SnQLFunction]:
                     ),
                     default_result_type="percentage",
                 ),
+                SnQLFunction(
+                    "group_array",
+                    # setting 10k as arbitraty upper limit for max_size argument, clickhouse limit is 100M
+                    required_args=[NumberRange("max_size", 0, 10000), ColumnTagArg("column")],
+                    snql_aggregate=lambda args, alias: CurriedFunction(
+                        "groupArray",
+                        [int(args["max_size"])],
+                        [args["column"]],
+                        alias,
+                    ),
+                    default_result_type="string",
```

Comment:
```suggestion
                    default_result_type="array",
```

Reply:
array doesn't seem to be a valid result type:
```
AssertionError: group_array: result type array not one of ['duration', 'day', 'percentage', 'date', 'kilobyte', 'kibibyte', 'megabyte', 'gigabyte', 'number', 'terabyte', 'nanosecond', 'microsecond', 'second', 'minute', 'hour', 'millisecond', 'exbibyte', 'petabyte', 'integer', 'pebibyte', 'exabyte', 'mebibyte', 'string', 'gibibyte', 'week', 'tebibyte', 'byte', 'bit']
```

should we create a new one?.

Reply:
Can you leave it as string with a `TODO: support the "array" type`, the typing is used by parts of the frontend to determine display so it doesn't impact what you're doing here